File: Backend/src/main/java/com/abdulrafy/backend/trading/entity/Trade.java
package com.abdulrafy.backend.trading.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "trades")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trade {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "portfolio_id", nullable = false)
    private UUID portfolioId;

    @Column(name = "asset_id", nullable = false)
    private UUID assetId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private OrderSide side;

    @Column(nullable = false, precision = 19, scale = 8)
    private BigDecimal quantity;

    @Column(nullable = false, precision = 19, scale = 4)
    private BigDecimal price;

    @Column(nullable = false, precision = 19, scale = 4)
    private BigDecimal fee;

    @Column(name = "idempotency_key", nullable = false, unique = true, length = 255)
    private String idempotencyKey;

    @Column(name = "executed_at", nullable = false)
    private Instant executedAt;
}

File: frontend/src/features/trading/types/trade.ts
export interface ExecuteTradeRequest {
  assetId: string;
  side: 'BUY' | 'SELL';
  quantity: number;
  idempotencyKey: string;
}

export interface TradeResponse {
  id: string;
  portfolioId: string;
  assetId: string;
  side: 'BUY' | 'SELL';
  quantity: number;
  price: number;
  fee: number;
  idempotencyKey: string;
  executedAt: string;
}
